src/util/OAuth2Util.js

```diff
@@ -55,14 +55,19 @@ define(['okta', './Enums', './Errors'], function (Okta, Enums, Errors) {
     _.extend(
       oauthParams,
       _.pick(options, 'clientId', 'redirectUri'),
-      _.pick(options.authParams, 'responseType', 'responseMode', 'display', 'scopes'),
+      _.pick(options.authParams, 'responseType', 'responseMode', 'display', 'scopes', 'state', 'nonce'),
       params
     );
 
     // Extra Options for Social Idp popup window title and id_token response timeout
     extraOptions.popupTitle = Okta.loc('socialauth.popup.title', 'login');
     extraOptions.timeout = options.oAuthTimeout;
 
+    _.extend(
```

Comment:
Yeah it's a bit weird that we have to do this (in general having options and extra options) - I think you've already created a jira for it, but if not can you create one?